packages/create-turbo/src/transforms/update-commands-in-readme.ts

```diff
@@ -0,0 +1,44 @@
+import path from "node:path";
+import fs from "node:fs/promises";
+import { TransformError } from "./errors";
+import type { TransformInput, TransformResult} from "./types";
+
+const meta = {
+  name: "update-commands-in-readme",
+};
+
+export async function transform(args: TransformInput): TransformResult {
+  const { prompts, example } = args;
+
+  const isOfficialStarter =
+    !example.repo ||
+    (example.repo.username === "vercel" && example.repo.name === "turbo");
+
+  if (!isOfficialStarter) {
+    return { result: "not-applicable", ...meta };
+  }
+
+  const selectedPackageManager = prompts.packageManager;
+  const readmeFilePath = path.join(prompts.root, "examples", "basic", "README.md");
+  try {
+    // Read the content of the file
+    let data = await fs.readFile(readmeFilePath, "utf8");
+
+    // an array of all the possible replacement strings.
+    const replacements = ['pnpm run', 'npm run', 'yarn run', 'bun run', 'pnpm', 'npm', 'yarn', 'bun'];
+    const replacementRegex = new RegExp(`\\b(?:${replacements.join('|')})\\b`, 'g');
+
+    // Replace all occurrences of regex with selectedPackageManager
+    data = data.replace(replacementRegex, `${selectedPackageManager} run`);
```

Comment:
Nice!

I think this could still be improved a little bit. Right now, there is no differentiation between code, and non code. So, for example in our design-system example, we have a section about Versioning & Publishing Packages that mentions npm as the npm registry. 

We don't want to change those as part of this transform. I think we should tighten the regex to only detect package managers that occur within backticks (or code blocks)


This will also fix occurrences within links (for example):
> This Turborepo uses [yarn v2+ (berry)](https://yarnpkg.com/) as a packages manager. It includes the following packages/apps:

This would be a great spot to add in some unit tests. I would recommend splitting the transform itself to it's own function for easier testing - something like:

`function replacePackageManager(packageManager, text) { ... }`

This will allow you to test it very easily without requiring any additional setup / teardown.

Reply:
As per your suggestion, I have refactored the code and the regex has been updated.
Below is the link for the test case I considered to validate the working of regex:
https://codefile.io/f/Svks01TyEk

I tried to consider the possible cases but in case I missed something or there is any scope for further improvement, please let me know.